include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -64,12 +64,22 @@ class format_bam : private detail::format_sam_base
     /*!\name Constructors, destructor and assignment
      * \{
      */
+#if SEQAN3_WORKAROUND_GCC_NO_CXX11_ABI
+    // string_buffer is of type std::string and has some problems with pre-C++11 ABI
+    format_bam() /*noexcept*/ = default; //!< Defaulted.
+    format_bam(format_bam const &) /*noexcept*/ = default; //!< Defaulted.
+    format_bam & operator=(format_bam const &) /*noexcept*/ = default; //!< Defaulted.
+    format_bam(format_bam &&) /*noexcept*/ = default; //!< Defaulted.
+    format_bam & operator=(format_bam &&) /*noexcept*/ = default; //!< Defaulted.
+    ~format_bam() /*noexcept*/ = default; //!< Defaulted.
+#else // ^^^ workaround / no workaround vvv
     format_bam() noexcept = default; //!< Defaulted.
     format_bam(format_bam const &) noexcept = default; //!< Defaulted.
     format_bam & operator=(format_bam const &) noexcept = default; //!< Defaulted.
     format_bam(format_bam &&) noexcept = default; //!< Defaulted.
     format_bam & operator=(format_bam &&) noexcept = default; //!< Defaulted.
     ~format_bam() noexcept = default; //!< Defaulted.
+#endif // SEQAN3_WORKAROUND_GCC_NO_CXX11_ABI
```

Comment:
Aren't defaulted constructors `noexcept` if they can?
Then we can just remove the `noexcept`.

Reply:
@smehringer Do you agree to remove noexcept?

@esiragusa If we do that I also need to change format_sam.

Either way is fine to me.

Reply:
Yes I agree. Explicitly defaulted ctors are constexpr and noexcept if they can be. To the best of my knowledge right now, there is no explicit reason for requiring noexcept (which would mean explicitly stating it here regardless of the defaults).